### ISSUE TITLE
Lexical affixes in polysynthetic languages

### ISSUE THREAD

**Issue by ftyers** [UniversalDependencies/docs#703]
This issue is related to issue #701. While that issue deals with the question of incorporating a noun into a verb, this issue talks about deriving a verb from a noun by means of a (sometimes formerly verbal) affix. This phenomenon is found in a number of languages, here I will cover three: Chukchi, Yupik and Greenlandic. In Yupik and Greenlandic these are called "postbases" and verbs that are formed from a noun + verbal postbase are called denominal verbs ([Malouf, 1999](https://pdfs.semanticscholar.org/e228/eae715ed56f5c8acf421cd5e1d2fd3c2a3b6.pdf)). 

In Chukchi a very similar phenomenon exists, from Dunn's grammar (§12.6)

> A number of languages have affixes which correspond to incorporating V[erb]’s in other languages. The Chukotko-Kamchatkan languages [...] also have small sets of derivational suffixes which, when added to N[oun]’s, function much like incorporating V’s (Bogoras 1922). They supply meanings such as ‘to fetch’, ‘to take off (clothing)’, ‘to put on (clothing)’, ‘to search for’, and ‘to consume, eat’. Suffixed to N’s, they derive V stems denoting unitary activities (Mithun, 1984)

In the Chukchi corpus I am working on, these appear in 15-20% of the texts I have annotated so far and are more frequent than nominal incorporation. I have come across the following affixes:

- More lexical:
   - `EAT`: Transitive, makes an intransitive verb where the object (thing being eaten) is filled.
   - `GO.TO`: Intransitive, makes an intransitive verb where the direction (place being gone to) is filled.
   - `DRAG.OUT`: Transitive, makes an intransitive verb when the object (thing being dragged) is filled.
   - `MAKE` (circumfix): Transitive, makes an intransitive verb where the object (thing being made) is filled.
   - `DES` (circumfix): Equivalent to "want to"
   - `SEARCH`: Transitive, makes an intransitive verb where the object (thing being searched for) is filled.
   - `CATCH`: Transitive, makes an intransitive verb where the object (thing being caught) is filled.

Although there are more described in the [glossing manual](http://chuklang.ru/static/chukchi_glosses_20171020.pdf).

One example (4) from [a story](http://chuklang.ru/media/texts/An%20evil%20spirit%20and%20a%20dicky%20bird.pdf) about an evil spirit and a bird:

![Captura de 2020-05-05 13-45-29](https://user-images.githubusercontent.com/449545/81067542-b8098a80-8ed6-11ea-8102-103048881918.png)
```
# text = Каԓьайӈын эгэ мынӄымԓёченмэв.
# text[phon] = kaɬʔajŋən eɣe mənqəmɬosenmew
# text[rus] = Злой дух [говорит]: «Ладно, давай есть друг у друга костный мозг».
# text[eng] = The evil spirit says: «Well, let us eat each other’s bone marrow».
1	Каԓьайӈын	Каԓьайӈын	NOUN	_	Case=Abs|Number=Sing	_	_	_	Gloss=злой.дух-AUG-NOM.SG
2	эгэ	_	ADV	_	_	3	advmod	_	Gloss=ладно
3	мынӄымԓёченмэв	_	VERB	_	_	1	parataxis	_	Gloss=1PL.S/A.SUBJ-костный.мозг-EAT-COMB-1PL.S/O.EXCL
4	.	.	PUNCT	_	_	1	punct	_	_
```

This is an intransitive verb, *мынӄымԓёченмэв* which translates as "let us eat each other’s bone marrow". The root is *ӄымԓ* [qəmɬ] "bone marrow", the "let us" is subjunctive `SUBJ`, while `COMB` can indicate reciprocity and `EAT` is a lexical affix that creates an intransitive verb from a noun X meaning to eat X. 

And another from [this text](http://chuklang.ru/media/texts/The%20day%20before%20yesterday.pdf):

![Captura de 2020-05-05 13-55-18](https://user-images.githubusercontent.com/449545/81068321-171bcf00-8ed8-11ea-925f-e2c9bdca0baf.png)
```
# text = Ӄԓявыԓ мургин гатапачпортӈыԓен.
# text[phon] = qɬawəɬ murɣin ɣatapasportŋəɬen
# text[rus] = DMP: Парень наш (с)делал паспорт.
# text[eng] = DMP: Our lad got a passport (lit. made).
1	Ӄԓявыԓ	Ӄԓявыԓ	NOUN	_	Animacy=Anim|Case=Abs|Number=Sing	3	nsubj	_	Gloss=мужчина-NOM.SG
2	мургин	мури	PRON	_	Possessive=Yes|Number=Plur|Person=1|PronType=Pers	1	nmod:poss	_	Gloss=мы-GEN
3	гатапачпортӈыԓен	_	VERB	_	Aspect=Perf	0	root	_	Gloss=PF-MAKE-паспорт-MAKE-PF.3SG
4	.	.	PUNCT	_	_	3	punct	_	_
```

Here the verb is an intransitive one "got a passport" formed of the noun *пачпорт* "passport" and the verbalising circumfix *та-STEM-ӈ*.

The Inuit languages have a similar structure as this example from Jacobsen's grammar of Yupik:

![Captura de 2020-05-05 14-04-40](https://user-images.githubusercontent.com/449545/81069138-69112480-8ed9-11ea-9abb-dcafa12745e0.png)

One could imagine an analysis such as:
```
# text = Mangteghaghrugllangllaghyunghitunga.
1       Mangteghaghrugllangllaghyunghitunga     mangtegha       NOUN    _       _       0       root    _       _
2       .       .       PUNCT   _       _       1       punct   _       _
```
where the lemma is "house", or
```
# text = Mangteghaghrugllangllaghyunghitunga.
1       Mangteghaghrugllangllaghyunghitunga     mangteghaghrugllangllaghyu       VERB    _       _       0       root    _       _
2       .       .       PUNCT   _       _       1       punct   _       _
```
where the lemma is "to want to build a big house". But I wonder if there might be a more adequate way of representing the information.

A similar example is found in [this story](http://chuklang.ru/media/texts/Doormat.pdf) in Chukchi, although combinations of these affixes are not frequent in the corpus I am annotating:
![Captura de 2020-05-05 15-47-00](https://user-images.githubusercontent.com/449545/81079635-b1374380-8ee7-11ea-8d1f-4f2367834d88.png)
```
# text = Гымнин ытԓьаӄай нырэԓгыӄанъявӈытаӈӄэн.
# text[phon] = ɣəmnin ətɬʔaqaj nəreɬɣəqanjawŋətaŋqen
# text[rus] = Моя мамочка хотела поехать на Первый (в Эгвекинот).
# text[eng] = My mum wanted to go to Egvekinot.
1       Гымнин  _       PRON    _       Animacy=Anim|Case=Gen|Number=Sing|Person=1|PronType=Pers        2       nmod:poss       _       Gloss=я-AN.GEN
2       ытԓьаӄай        ытԓьаӄай        NOUN    _       Animacy=Anim|Case=Abs|Deriv=Dim|Number=Sing     3       nsubj   _       Gloss=мать-DIM-NOM.SG
3       нырэԓгыӄанъявӈытаӈӄэн   _       PROPN    _       _       0       root    _       Gloss=ST-DES-белый-ущелье-GO.DO-DES-ST.3SG
4       .       .       PUNCT   _       _       3       punct   _       _
```

Here *Эԓгыӄанъяв* is a place name and `GO.DO` can be used with other placenames such as *тынотапынмынтагъак* "I went to Nutepelmen" < *Нутэпылмын* "[Nutepelmen](https://en.wikipedia.org/wiki/Nutepelmen)".

In his dependency analysis of Greenlandic, principally targetted at Greenlandic→Danish machine translation, [Eckhard Bick (2019)](https://corpora.linguistik.uni-erlangen.de/data/konvens/proceedings/papers/KONVENS2019_paper_36.pdf), these suffixes are split off as separate nodes in the tree. Among the reasons he gives is the fact that subunits can have word-external arguments, although he argues against over-splitting.

> While splitting off of incorporated arguments,auxiliaries   and   light   adverbs   clearly   pushes syntax under the water-line of the word boundary and helps to create a deeper syntax and a more universal   dependency   tree,   there   is   also   the danger of splitting off morphemes that are less syntactic in nature and part of larger semantic lexical units. For instance, in our example, theword   for   plane   can   be   morphologically deconstructed into the root *timmi* (plane) and the affixes  TAR (uses to)  and  TUQ  (that which), literally meaning something (or somebody) that uses to fly. However, such a deconstruction is only   of   etymological   interest,   there   are   no external syntactic reasons for this (such as the existence of `@i->V` arguments), and the lexical minimal unit in terms of object equivalence in the real world is clearly  plane.  Similarly, the verb   root  *titartaavoq*   (draw)  is   originally decomposed by the FST as  *titartar(paa)+HTR*, i.e. with a transitive root and an affix denoting "half-transitivity" (i.e. taking an indefinite object in instrumental case). However, the  HTR  affix,while   leaving   morphological   traces,   does   not correspond to a syntactic node, and since the external object is in an oblique case rather than ordinary object case (absolute), it syntactically "prefers" the longer and already half-transitive form  *titartaavoq*  as   its   dependency   head   (i.e.with HTR included).

As far as UD is concerned, the surface syntax is clear, as with #701, these are largely intransitive verbs. But simply treating them as such belies argument structure relations that are going on underneath. For example, we would be relegated to having a separate verb for every object that can be made, eaten or gone to, and that information would not be available without further processing. 

We could produce the missing "predicate" in the enhanced representation, much like we do with predicate ellipsis, e.g.
![Captura de 2020-05-05 17-43-10](https://user-images.githubusercontent.com/449545/81092217-332f6880-8ef8-11ea-8ed0-ec05133e5afe.png)
```
# text = Ӄԓявыԓ мургин гатапачпортӈыԓен.
# text[phon] = qɬawəɬ murɣin ɣatapasportŋəɬen
# text[rus] = DMP: Парень наш (с)делал паспорт.
# text[eng] = DMP: Our lad got a passport (lit. made).
1       Ӄԓявыԓ  Ӄԓявыԓ  NOUN    _       Animacy=Anim|Case=Abs|Number=Sing       3       nsubj   3.1:nsubj       Gloss=мужчина-NOM.SG
2       мургин  мури    PRON    _       Possessive=Yes|Number=Plur|Person=1|PronType=Pers       1       nmod:poss       _       Gloss=мы-GEN
3       гатапачпортӈыԓен        _       VERB    _       Aspect=Perf|Number[subj]=Sing|Person[subj]=3|Valency=1|VerbForm=Fin     0       root    _       Gloss=PF-MAKE-паспорт-MAKE-PF.3SG
3.1     _       _       VERB    _       Predicate=Make|Valency=2        _       _       0:root  Gloss=MAKE
3.2     пачпорт пачпорт NOUN    _       _       _       _       3.1:obj Gloss=паспорт
4       .       .       PUNCT   _       _       3       punct   _       _
```

The motivation behind encoding the morpheme as `Predicate=Make` is that there should be a finite set, and it avoids having "technical" surface forms or lemmas, as *make* is a circumfix, *та-STEM-ӈ*.

I think that this phenomenon should be taken into account in the ongoing discussions about enhanced UD and predicate-argument structure. I would welcome any thoughts or comments too.

---

1. Michael Dunn (1999) *Grammar of Chukchi*. PhD Thesis
2. Bick, E. (2019). "Dependency Trees for Greenlandic". *Proceedings of the 15th Conference on Natural Language Processing (KONVENS 2019)*. 140-148
3. Robert Malouf (1999) "West Greenlandic noun incorporation in a monohierarchical theory of grammar." In Gert Webelhuth, Andreas Kathol, and Jean-Pierre Koenig (ed.), *Lexical and Constructional Aspects of Linguistic Explanation*.  47-62.  
4. Marion Mithun (1984). "The Evolution of Noun Incorporation". *Language* 60,4: 847-
894
5. Steven Jacobson (2001) *A Practical Grammar of the St. Lawrence Island / Siberian Yupik Eskimo Language*

**Comment by Stormur** [comment]
I think that this might be the best solution, but just a question: why annotate the `Predicate=Make` information only as an enhanced dependencies? From what you say it is something that seems to belong more to the analysis of lexical derivation, but still I think that it might deserve its "ordinary" feature, just as the `NounClass` feature for Bantu, especially if this group is finite. And second question: is it not possible to derive a lemma for the form гатапачпортӈыԓен? One could perhaps just use the noun's lemma as the lemma, which is shown to having been verbalised by a feature like `VerbAffix`. I am saying this because I see it as being not so different (the exact opposite, actually) from the analysis, e.g. in Latin, of a verbal nominal form as having a verb as its lemma, for example:

`1     doctissimus doceo VERB    _       ...|Degree=Abs|...|**VerbForm=Part**|...       _       _       _ Gloss=most learned`

So I would suggest something like:

```
# Ӄԓявыԓ мургин гатапачпортӈыԓен.
# text[phon] = qɬawəɬ murɣin ɣatapasportŋəɬen
# text[rus] = DMP: Парень наш (с)делал паспорт.
# text[eng] = DMP: Our lad got a passport (lit. made).
1       Ӄԓявыԓ  Ӄԓявыԓ  NOUN    _       Animacy=Anim|Case=Abs|Number=Sing       3       nsubj   _      Gloss=мужчина-NOM.SG
2       мургин  мури    PRON    _       Possessive=Yes|Number=Plur|Person=1|PronType=Pers       1       nmod:poss       _       Gloss=мы-GEN
3       гатапачпортӈыԓен        пачпорт       **NOUN**    _       Aspect=Perf|Number[subj]=Sing|Person[subj]=3|Valency=1|VerbForm=Fin|**VerbAffix=Make**     0       root    _       Gloss=PF-MAKE-паспорт-MAKE-PF.3SG
4       .       .       PUNCT   _       _       3       punct   _       _
```

With no need for particular enhanced dependencies.

To sum it up: this would mean that, as in Latin we have an adjective (_doctissimus_) displaying nominal features (degree, case, gender,...) which is "reduced" to a `VERB` (_doceo_) because of regular morpholexical processes, so here in Chukchi we have a verb (_гатапачпортӈыԓен_) which is "reduced" to a `NOUN` (_пачпорт_), again due to regular processes (a finite class of verbalising affixes), but whose form displays verbal features. Two symmetrical phenomena!

As for `VerbForm`, the UD guidelines say:

> Even though the name of the feature seems to suggest that it is used exclusively with verbs, it is not the case. Some verb forms in some languages actually form a gray zone between verbs and other parts of speech (nouns, adjectives and adverbs).

**Comment by dan-zeman** [comment]
While indeed interesting and potentially useful for language understanding applications, this seems to be largely about word formation and thus beyond the scope of Universal Dependencies. I agree that capturing the word-internal structure may be more useful in certain languages (not just the polysynthetic ones; long compounds in German would benefit from internal analysis too) but still it feels like a target of a separate project built outside UD or on top of UD (e.g. [Universal Derivations](https://ufal.mff.cuni.cz/universal-derivations)).

Unlike @Stormur I don't think that _гатапачпортӈыԓен_ should be tagged `NOUN`. If it is a verb derived from a noun, it should be tagged `VERB` and preferably get a verbal lemma (_тапачпортӈ_?) It does not seem to be a `VerbForm=VNoun` to me. And I would expect a `VERB` to be the root of a finite clause.

However, are there examples where subword units have arguments outside the word, as Eckhard Bick claims for Greenlandic? That could be taken as evidence that the orthographic word actually consists of multiple syntactic words, and a MWT analysis could be proposed.

**Comment by ftyers** [comment]
I will be a bit of a devil's advocate here... So the suggestion for the Greenlandic example would be to have a tree like:
```
# text = Mangteghaghrugllangllaghyunghitunga.
1       Mangteghaghrugllangllaghyunghitunga     mangteghaghrugllangllaghyu       VERB    _       Mood=Ind|Number=Sing|Person=1       0       root    _       _
2       .       .       PUNCT   _       _       1       punct   _       _
```
I'm not sure how well this does on the 6 criteria:

* UD needs to be satisfactory for analysis of individual languages.
   - I'm not sure how satisfactory it is, but it's barely an analysis.
* UD needs to be good for linguistic typology.
   - Perhaps it's ok to show that "Greenlandic has long words"
* UD must be suitable for rapid, consistent annotation.
   - Definitely scores high on this criterion
* UD must be suitable for computer parsing with high accuracy.
   - Definitely scores high on this criterion, just make everything `root`
* UD must be easily comprehended and used by a non-linguist.
   - It probably does well on this, but then there is not much to comprehend in the analysis.
* UD must provide good support for downstream NLP tasks.
   - I'm not sure that this helps any downstream task at all

It could be that what is going on in Greenlandic isn't analysable with UD, but I think that would be a bit of a shame, and not very "universal". You could say that some of the information could go into the morphological inflectional features, but then the problem is that they are not ordered, so the recoverability from word order that you have when ignoring order in word-based auxiliaries does not apply (as we do not record the order of the features).

Regarding the Chukchi example,

- It's a verb derived from a noun, the noun is "паспорт" and the verbal/postbase circumfix is *та-  -ӈ*. The finite inflectional affixes then are *га-* and *-ԓен*, which (together) denote 3rd person singular non-imperfect non-future. It is a finite form.
- There are examples of external arguments, among others in possessor raising, where the possessor of a noun is promoted to object when the noun is incorporated. 
   - e.g. They read my paper → They paper-read mine.
- The Malouf article also has some examples

Regarding another project, the UDeriv project looks interesting, but it's not exactly what I'm looking at. Perhaps a project like "Polysynthetic Dependencies" would work, that builds on top of UD and produces useful trees for these languages. But, it would have essentially the same aims, and would end up to a fragmenting and duplicating work (a whole new repo, website, scripts etc.), exactly for the groups of languages that don't dispose of enormous amounts of resources for getting things done in the first place.

**Comment by dan-zeman** [comment]
I totally agree that it would be a shame if we have to say that UD is not able to analyze these languages (or if it is only capable of providing an analysis significantly shallower than in other languages). But it seems difficult to design guidelines for word-internal structure without betraying the lexicalist hypothesis, which is one of the pillars that we usually stress when introducing UD.

Perhaps we could do something similar to what we do with light-verb constructions. In languages like English, constructions like _to take a picture of something_ are given a "normal" syntactic analysis. But in languages like Persian and Hindi, where LVC are widespread and sort of a feature of the grammar, the `compound:lvc` relation is used. So here we could say that in languages like Chukchi, Yupik and Greenlandic, analysis of what appears to be word-internal structure is done, but limited to well described types of constructions?

The examples like "They paper-read mine" are definitely a good evidence why one would want to treat the long word as several syntactic words (even when the external modifier is not present!), but like Eckhard, I would advocate for not overdoing it (not sure how exactly the line should be drawn; separating a verb from its arguments is the first step, maybe separating "big" from "house" should be allowed as well...)

**Comment by ftyers** [comment]
I don't think we need to betray the lexicalist hypothesis, it is also possible for syntactic operations to happen "in the lexicon" and for the outcome of syntactic operations to be stored "in the lexicon" (UD allows this through the `flat` and `fixed` relations). 

If we take the weak hypothesis, then inflection is morphology that is relevant to syntax (agreement etc.) and derivation is that which isn't. In the above example "make a passport", the "make" part is relevant to the external syntax, e.g. the argument structure of the whole clause, what tells you that "he-ABS" is a subject, not an object.

These lexical affixes could be treated as allomorphs of regular words in the lexicon, e.g. in Chukchi, the 'EAT' affix is -u-, while there is a verb 'to eat' which is -Ru, with an initial r/n alternation. 

I completely agree that dependency analysis of what happens inside words should be very restricted. I would also like to avoid it in the basic representation as much as possible. So I wonder if the enhanced dependencies are the right place for it. As you mention, if the typological features result in a shallow analysis in the basic representation, perhaps it could be made up for in the enhanced representation? 

I agree that a good place to start would be the verb/arguments, basically anything that has an effect on clause structure. Regarding "big house", I am not sure, potentially it could be accounted for with a feature on house, e.g. "AUG" (augmentative), it depends on if the word comes from a finite set, or if any adjective modifier can go there. 

A couple more notes, regarding possessor raising, here is what Dunn has to say:

> 12.2.3 Possessor raising
> A possessed noun in the absolutive case can be incorporated without reduction of valency, with the possessor filling the S/O slot of the verb (intransitive verbs seem to have to be undergoer subject intransitives, i.e. S o verbs, not S a ). This type of incorporation only occurs when the possessed noun is inalienably possessed. Note however that ‘inalienable possession’ is not otherwise indicated as a grammatical category in Chukchi; usually the inalienably possessed noun is a body part (see 019 for a possible exception).


In Malouf's work below, (5) shows two ergatives, one is the possessor of the noun, the other is the subject of the clause
![Captura de 2020-05-09 14-56-27](https://user-images.githubusercontent.com/449545/81475627-4e0a2180-9205-11ea-9652-0a50022697cf.png)

**Comment by dowobeha** [comment]
> However, are there examples where subword units have arguments outside the word, as Eckhard Bick claims for Greenlandic? That could be taken as evidence that the orthographic word actually consists of multiple syntactic words, and a MWT analysis could be proposed.

[Yes](https://universal.grew.fr/?custom=6a52d91590c67)

```
# sent_id = Jacobson.Ch9.gold:21
# text = Mekelghiighem qikmimun aghnaghaq qilugtestaa.
# gloss = The boy let the dog bark at the girl.
# labels = complete-dep anno-hayley
1-2	Mekelghiighem	_	_	_	_	_	_	_	Analysis=mekelghiigh(N)^[Rel.Sg]|Gloss=the-boy
1	Mekelghiigh	mekelghiigh	NOUN	_	_	7	nsubj	_	Analysis=mekelghiigh(N)|Gloss=boy
2	em	em	X	_	Case=Erg|Number=Sing	1	dep:infl	_	Analysis=[Rel.Sg]
3-4	qikmimun	_	_	_	_	_	_	_	Analysis=qikmigh(N)^[All.Sg]|Gloss=the-dog
3	qikmi	qikmi	NOUN	_	_	6	nsubj	_	Analysis=qikmigh(N)|Gloss=dog
4	mun	mun	X	_	Case=All|Number=Sing	3	dep:infl	_	Analysis=[All.Sg]
5	aghnaghaq	aghnaghaq	NOUN	_	Case=Abs|Number=Sing	6	obj	_	Analysis=aghnaghagh*(N)^[Abs.Sg]|Gloss=the-girl
6-9	qilugtestaa	_	_	_	_	_	_	_	Analysis=qilug(V)^+(te)ste(CmpdVbl)^[Ind.Trns]^[3Sg.3Sg]|Gloss=he-let-it-bark-at-her|SpaceAfter=No
6	qilug	qilug	VERB	_	_	7	xcomp	_	Analysis=qilug(V)|Gloss=to-bark
7	test	test	VERB	_	_	0	root	_	Analysis=+(te)ste(CmpdVbl)|Gloss=to-let-one-V
8	a	a	X	_	Mood=Ind|Subcat=Tran	7	dep:infl	_	Analysis=[Ind.Trns]
9	a	a	X	_	Number[obj]=Sing|Number[subj]=Sing|Person[obj]=3|Person[subj]=3	7	dep:infl	_	Analysis=[3Sg.3Sg]
10	.	.	PUNCT	_	_	7	punct	_	Analysis=.
```

This could go even farther. For example, the following sentence in Yupik would have a single orthographic verb composed of a verb stem (to bark), a derivational morpheme (to let V), a derivational morpheme (to want to V), and finally an inflection.

| The man wanted the boy to let the dog bark at the girl.

* The root morpheme *to bark* would have *dog* and *girl* as its arguments.
* The derivational morpheme *to let (something) V* would have *boy* as its argument
* The derivational morpheme *to want to V* would have *man* as its argument
* The inflectional morpheme would be associated with the derived verb *to want to let (something) bark*


Here's another similar example from Central Alaskan Yup'ik (Miyaoka 2012)

<img width="439" height="151" alt="Image" src="https://github.com/user-attachments/assets/59703482-193c-429f-b886-ada7b17f58ef" />

In the version of (96a) that includes "mother":

 | He let my mother singe the grass

* The root morpheme *to singe* would have *mother* and *grass* as arguments
* The derivational morpheme *to let (something) V* would have *he* (if it was explicitly present) as its argument
* The inflectional morpheme would be associated with the derived verb *to let (someone) singe (something)*

**Comment by dowobeha** [comment]
@nschneid @ftyers @dan-zeman 

Should this be re-opened? Or should I start a new issue?
